tests/config/test_find_default_config_files.py

```diff
@@ -0,0 +1,68 @@
+# Licensed under the GPL: https://www.gnu.org/licenses/old-licenses/gpl-2.0.html
+# For details: https://github.com/PyCQA/pylint/blob/main/LICENSE
+# Copyright (c) https://github.com/PyCQA/pylint/blob/main/CONTRIBUTORS.txt
+
+
+import pytest
+
+from pylint.config.find_default_config_files import _cfg_has_config, _toml_has_config
+
+
+@pytest.mark.parametrize(
+    "content,expected",
+    [
+        ["", False],
+        ["(not toml valid)", False],
+        [
+            """
+[build-system]
+requires = ["setuptools ~= 58.0", "cython ~= 0.29.0"]
+""",
+            False,
+        ],
+        [
+            """
+[tool.pylint]
+missing-member-hint = true
+""",
+            True,
+        ],
+    ],
+)
+def test_toml_has_config(content, expected, tmp_path):
+    """Test that a toml file has a pylint config."""
+    fake_toml = tmp_path / "fake.toml"
+    with open(fake_toml, "w", encoding="utf8") as f:
+        f.write(content)
+    assert _toml_has_config(fake_toml) == expected
+
+
+@pytest.mark.parametrize(
+    "content,expected",
+    [
+        ["", False],
```

Comment:
We do not handle parsing error in setup.cfg, so pylint will crash if a setup.cfg is malformed. (We had the issue for toml, and we handle it.)

Reply:
I'm still not really sure if we should even allow `pylint` to run with malformed `setup.cfg` and `pyproject.toml`'s. I know we have added that new message but looking at it again it seems kind of weird. We're not a `.toml` linter so raising issues about them will always be a bit strange. Why don't we require users to fix their bad configuration before their tools can run?

Reply:
The mere existence of a malformed setup.cfg / pyproject.toml would crash pylint here even if we do not use it at all. This function is searching for a pylint section in a configuration, but the pylintrc could exists. If the malformed setup.cfg / or pyproject.toml is the main configuration file then I think we should report the issue immediately and exit.

Reply:
+1 for passing here.

But do we then agree that the `config-parse-error` message should be removed?

Reply:
> Why don't we require users to fix their bad configuration before their tools can run?

I think it's because we did not fix our badly designed and documented configuration files (yet) 😄 so error messages helps in the meantime. Removing the need to have section header and having clear error message for option that do not exists or have impossible values, and a good documentation would be better.

Reply:
Okay, let's pick this up in the Parser refactor then.

Lots to take into account there still 😄

Reply:
Yeah maybe looking at all the issues talking about configuration would be a (labor-intensive) good start. I'm going to do that.

Reply:
@Pierre-Sassoulas You can look at the migration project I openend. I think I put all relevant issues in there already.